restli-int-test/src/test/java/com/linkedin/restli/examples/TestRestLiValidation.java

```diff
@@ -557,12 +555,18 @@ public void testPartialUpdateFailure(RestClient restClient, Object builder, Stri
             "{\"patch\": {\"validationDemoNext\": {\"$set\": {\"stringA\": \"some value\"}}}}",
             // A field (MapWithTyperefs/key1) containing a CreateOnly field (MapWithTyperefs/key1/id) has to be partially set
             "{\"patch\": {\"MapWithTyperefs\": {\"key1\": {\"$set\": {\"message\": \"some message\", \"tone\": \"SINCERE\"}}}}}",
-            // Okay to set a field containing a ReadOnly field by omitting the ReadOnly field
+            // Okay to set a field containing a ReadOnly field if the ReadOnly field is omitted
             "{\"patch\": {\"$set\": {\"ArrayWithInlineRecord\": [{\"bar2\": \"missing bar1\"}]}}}",
+            "{\"patch\": {\"$set\": {\"UnionFieldWithInlineRecord\": {\"com.linkedin.restli.examples.greetings.api.myRecord\": {}}}}}",
+            "{\"patch\": {\"$set\": {\"validationDemoNext\": {\"stringA\": \"no stringB\"}}}}",
             // Okay to delete a field containing a ReadOnly field
             "{\"patch\": {\"$delete\": [\"ArrayWithInlineRecord\"]}}",
             // Okay to delete a field containing a CreateOnly field
-            "{\"patch\": {\"MapWithTyperefs\": {\"$delete\": [\"key1\"]}}}"
+            "{\"patch\": {\"MapWithTyperefs\": {\"$delete\": [\"key1\"]}}}",
+            // Okay to set a ReadOnly field if it's the descendant of an array
+            "{\"patch\": {\"$set\": {\"ArrayWithInlineRecord\": [{\"bar1\": \"setting ReadOnly field\", \"bar2\": \"foo\"}]}}}",
```

Comment:
What does the dollar sign `$` represent in this string-formed request?

Reply:
This is the request body of the `PARTIAL_UPDATE` request. See here for some details of the exact protocol: https://linkedin.github.io/rest.li/spec/protocol#partial-update